tests/pipeline/test_numerical_expression.py

```diff
@@ -128,6 +143,64 @@ def test_validate_bad(self):
         with self.assertRaises(TypeError):
             (f > f) > f
 
+    def test_combine_datetimes(self):
+        with self.assertRaises(TypeError) as e:
+            self.d + self.d
+        message = e.exception.args[0]
+        expected = (
+            "Don't know how to compute datetime64[ns] + datetime64[ns].\n"
+            "Arithmetic operators are only supported on Factors of dtype "
```

Comment:
You probably don't want calendar date arithmetic there though.  I added BusinessDaysUntil{Next,Previous}Earnings for that use case.